### PR TITLE
404 on static assets

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

Changes `curl --head http://localhost:8000/static/chunk-NONO.js` from:

```http
HTTP/1.1 302 Found
Date: Tue, 09 Nov 2021 22:27:19 GMT
Server: WSGIServer/0.2 CPython/3.9.7
Content-Type: text/html; charset=utf-8
Location: /login?next=/static/chunk-NONO.js
X-Frame-Options: DENY
Content-Length: 0
Vary: Cookie
X-Content-Type-Options: nosniff
Referrer-Policy: same-origin
```

to

```http
HTTP/1.1 404 Not Found
Date: Tue, 09 Nov 2021 22:30:31 GMT
Server: WSGIServer/0.2 CPython/3.9.7
Content-Type: text/html; charset=utf-8
X-Frame-Options: DENY
Content-Length: 21
Vary: Cookie
X-Content-Type-Options: nosniff
Referrer-Policy: same-origin
```

Static files that exist still get served... I assume [directly by django](https://stackoverflow.com/questions/14408017/does-django-staticfiles-skip-the-middleware), so only serving files that were not there changed.

## Why?

I found a issue with edge caching `/static/chunk-*.js` files that were not there. 

- If you requested `/static/chunk-NOPE.js` and were logged out, you'd get a 302 to the login page. Bad, but fine compared to the alternative.
- If you requested `/static/chunk-STILL-NOPE.js`, and were logged in, you'd get a `200` response with some HTML. Cloudflare would then edge cache this file for a day, so that whoever else later tried opening `chunk-STILL-NOPE.js`, would get some HTML.

Basically, request a file too early (could this happen?) and the file is broken for everyone. 

## Proof of concept

Try this, `curl --head https://app.posthog.com/static/testing-CHANGEME.js` and see a `302`. Visit the same page in the browser and later the same command responds with a `200`

## How did you test this code?

Tested locally by making requests both before and after fixes. I assume this behaviour is consistent with how production does it, but I'm not 100% sure.

### CODE REVIEW

**Review by fuziontech — APPROVED** [review]
I'm legit surprised by that `302` from Django. Surprised and disappointed

**Comment by hit2hat** [comment]
Try to access app.posthog.com from any new browser and u will be surprised by that it’s a blank page and many chunks not loaded (404). Fix please…

**Comment by hit2hat** [comment]
it’s can be helpful

01028A96-0849-48F5-8635-784345BF6591

**Comment by hit2hat** [comment]
@fuziontech

**Comment by fuziontech** [comment]
Thanks for the @ mention @hit2hat 

Just put in a revert for this https://github.com/PostHog/posthog/pull/7005

**Comment by fuziontech** [comment]
@hit2hat Fixed! Cloudflare was caching our 404's extremely aggressively so had to bust the cache. Sorry for the inconvenience here!
